config/operator.yaml

```diff
@@ -59,8 +59,33 @@ spec:
             periodSeconds: 1
             successThreshold: 1
             timeoutSeconds: 15
+          securityContext:
+            runAsUser: 65532
+            runAsGroup: 65532
+            runAsNonRoot: true
+            readOnlyRootFilesystem: true
+            seccompProfile:
+              type: RuntimeDefault
+            allowPrivilegeEscalation: false
+            capabilities:
+              drop:
+                - ALL
           terminationMessagePolicy: FallbackToLogsOnError
+          volumeMounts:
+            - mountPath: /tmp
```

Comment:
I had a failure without it:

```
$ k get pod
NAME                                READY   STATUS             RESTARTS      AGE
spicedb-operator-5b666bff7-kfbtt    1/1     Running            0             39m
spicedb-operator-7d9bf5b5b7-8vv8t   0/1     CrashLoopBackOff   6 (37s ago)   6m30s
```

I don't have the logs anymore but here's what I saw as captured by Cloud Logging:

![image](https://github.com/authzed/spicedb-operator/assets/52710/e275c57d-9b57-49f2-8f61-6f638a305db7)

Here's the full message:

```json
{
  "textPayload": "panic: mkdir /tmp/kubebuilder-envtest: read-only file system\n\ngoroutine 1 [running]:\nsigs.k8s.io/controller-runtime/pkg/internal/testing/addr.init.0()\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.1/pkg/internal/testing/addr/manager.go:57 +0x165",
  "insertId": "2walpd5izxoo3b2x",
  "resource": {
    "type": "k8s_container",
    "labels": {
      "cluster_name": "cloud2-dev-green",
      "location": "us-east1",
      "pod_name": "spicedb-operator-7d9bf5b5b7-8vv8t",
      "container_name": "spicedb-operator",
      "project_id": "prefect-dev-cloud2",
      "namespace_name": "spicedb-operator"
    }
  },
  "timestamp": "2023-08-16T20:47:11.609563626Z",
  "severity": "ERROR",
  "labels": {
    "k8s-pod/app": "spicedb-operator",
    "k8s-pod/app_kubernetes_io/instance": "spicedb-operator",
    "k8s-pod/pod-template-hash": "7d9bf5b5b7",
    "compute.googleapis.com/resource_name": "gke-cloud2-dev-green-application-3cc5-79a27aea-bj2k"
  },
  "logName": "projects/prefect-dev-cloud2/logs/stderr",
  "receiveTimestamp": "2023-08-16T20:47:33.420653682Z"
}
```

Reply:
Hm, envtest shouldn't be running at all outside of the e2e tests. I think we should look at this a bit more.

Reply:
I'm guessing it's this: https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/internal/testing/addr/manager.go#L44-L59

Reply:
lmaooo good 'ol global init functions.

well, in the interest of not letting the best be the enemy of the good, I'd propose we either:

* Merge with the /tmp volume; or,
* Remove the readOnlyRootFilesystem restriction

to unblock this change, then explore whatever controller-runtime is doing with the upstream maintainers

Reply:
One downside of the /tmp volume is that it can cause issues with Cluster Autoscaler, since CA won't evict pods that have volumes attached

https://github.com/kubernetes/autoscaler/issues/3984

Reply:
I have https://github.com/authzed/controller-idioms/pull/40 which removes controller-runtime from controller-idioms, which I will then pull into https://github.com/authzed/spicedb-operator/pull/243 this PR which removes the spicedb-operator dependency.

Once we get those in, there should be no need for a filesystem anymore 🎉

Reply:
What's the best way for us to proceed? I can update this to remove the volumes, but I have no easy way of testing

You should also be able to push to my branch

Reply:
Can you rebase this branch on `main` and remove the tmp directory? I tested it in a local kind cluster and it seems to work fine.

I can do those steps for you if you prefer, but I like keeping author/committer the same when possible 😄

Reply:
Rebased and removed!